pyanaconda/modules/payloads/source/nfs/nfs.py

```diff
@@ -46,6 +47,20 @@ def for_publication(self):
         """Return a DBus representation."""
         return NFSSourceInterface(self)
 
+    def process_kickstart(self, data):
+        """Process the kickstart data."""
+        nfs_url = create_nfs_url(data.nfs.server, data.nfs.dir, data.nfs.opts)
+        self.set_url(nfs_url)
+
+    def setup_kickstart(self, data):
+        """Setup the kickstart data."""
+        (opts, host, path) = parse_nfs_url(self.url)
```

Comment:
This is out of scope of this PR, but shouldn't we replace the URL with a DBus structure?

Reply:
I don't think so. It's easier to parse and merge the url so there is no need to complicate an API here. From my PoV.
Also we're doing the same with proxy strings.